pkg/collect/dns.go

```diff
@@ -104,14 +158,17 @@ func getKubernetesClusterIP(client kubernetes.Interface, ctx context.Context) (s
 	return service.Spec.ClusterIP, nil
 }
 
-func troubleshootDNSFromPod(client kubernetes.Interface, ctx context.Context) (string, error) {
+func troubleshootDNSFromPod(client kubernetes.Interface, ctx context.Context, nonResolvableDomain string, image string) (string, error) {
 	namespace := "default"
-	command := []string{"/bin/sh", "-c", `
-		set -x
+	command := []string{"/bin/sh", "-c", fmt.Sprintf(`
```

Comment:
Is there a reason why you switched from `nslookup` to `dig`? Images such as `busybox` and `alpine` package `nslookup`.

It might be slightly more code, but you can attempt `dig` and if the binary is not present, try `nslookup`. The collector will be more robust this way.

Reply:
The `nslookup` command in busybox is not working correctly

E.g.

```bash
k run -it --rm debug --image busybox -- /bin/sh
If you don't see a command prompt, try pressing enter.
/ #
/ # nslookup kubernetes.default
Server:         10.43.0.10
Address:        10.43.0.10:53

** server can't find kubernetes.default: NXDOMAIN
```

I've changed to `dig` simply because of the base image has `dig` installed and the `dig +short` is easier to parse.

Reply:
Could you please add this as a comment in the code for future context on the technical choice?

Reply:
many thanks @banjoh! I'll update the docs accordingly as well.